### ISSUE TITLE
[audio] standardize conv output length computation

### ISSUE THREAD

**Issue by eustlb** [huggingface/transformers#42118]
As discussed offline with @ArthurZucker , there's a lot f audio models that require knowledge off conv1d output length (eg for mask length computation). For this reason, exact formula for such is duplicated and rewritten at each integration... let's standardize!

**Comment by diegoakel** [comment]
Hello @eustlb my PR seems ok from the tests. I used the generalized version. I think I didn't forget any model.

```python
def _conv_out_length(input_length, kernel_size, stride, pad=0, dilation=1):
    # 1D convolutional layer output length formula taken
    # from https://pytorch.org/docs/stable/generated/torch.nn.Conv1d.html
    return torch.div(input_length + 2 * pad - dilation * (kernel_size - 1) - 1, stride, rounding_mode="floor") + 1
```
